From 9ed7933d4dd13f201d0f8a9837892e22a09394cb Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:15:52 -0400
Subject: [alsa] oxygen: Xonar DG(X) modify high-pass filter control

Message-id: <140972823573-64-patchset-jkysela@redhat.com>
Patchwork-id: 90764
O-Subject: [RHEL 7.1 PATCH 063/442] ALSA: oxygen: Xonar DG(X): modify high-pass filter control
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Change the 'put' function of the high-pass filter control to use the new
SPI functions.

Signed-off-by: Roman Volkov <v1ron@mail.ru>
Signed-off-by: Clemens Ladisch <clemens@ladisch.de>

(picked from kernel: fc114e9fbaf555e2d7fbfe144dac716142e22331)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/oxygen/xonar_dg_mixer.c b/sound/pci/oxygen/xonar_dg_mixer.c
index 41ee393..dfdfc99 100644
--- a/sound/pci/oxygen/xonar_dg_mixer.c
+++ b/sound/pci/oxygen/xonar_dg_mixer.c
@@ -325,6 +325,8 @@ static int input_sel_put(struct snd_kcontrol *ctl,
  return changed;
 }
 
+/* ADC high-pass filter */
+
 static int hpf_info(struct snd_kcontrol *ctl, struct snd_ctl_elem_info *info)
 {
  static const char *const names[2] = { "Active", "Frozen" };
@@ -354,8 +356,10 @@ static int hpf_put(struct snd_kcontrol *ctl, struct snd_ctl_elem_value *value)
  if (value->value.enumerated.item[0])
   reg |= CS4245_HPF_FREEZE;
  changed = reg != data->cs4245_shadow[CS4245_ADC_CTRL];
- if (changed)
-  cs4245_write(chip, CS4245_ADC_CTRL, reg);
+ if (changed) {
+  data->cs4245_shadow[CS4245_ADC_CTRL] = reg;
+  cs4245_write_spi(chip, CS4245_ADC_CTRL);
+ }
  mutex_unlock(&chip->mutex);
  return changed;
 }
-- 
1.7.1